src/server/api/router/dbRouter/students/utils.ts

```diff
@@ -35,6 +53,17 @@ export const getWhereObjFromFilters = (filters: TableFilters['get']): Prisma.Stu
   return { OR: output };
 };
 
+export const getWhereObjFromFiltersAndStatus = (filters: TableFilters['get']): Prisma.StudentWhereInput => {
+  let output: Prisma.StudentWhereInput['AND'] = [];
+
+  if (filters.search) output = [...output, { OR: searchFilters(filters.search) }];
+  if (filters.licenseFileStatus.length > 0) {
+    const licenseFileStatuses = licenseFilesFilter(filters.licenseFileStatus);
+    output = [...output, { licenseFiles: { some: { status: { in: licenseFileStatuses } } } }];
```

Comment:
The Logic here is incorrect. There is a difference between the "current student license files status" and "all the previous statuses of the student".

A student might have an active license file, and a failed license file in the past. The failed one should not be picked up here, instead we should filter by the most recent license file status.

To help you out, please check the logic we did to even decide on which license file to show in the list students data table. Check the following file: `@/server/api/router/dbRouter/students/query.ts`

We use the `getStudentStatusFromLicenseFiles` util to check for it, you might need that here too.